### DISCUSSION TITLE
`gas` override ignored in contract object's built-in `write` action

### DISCUSSION THREAD

**Discussion by firnprotocol** [wevm/viem#1775] [General]
consider the following code:
```javascript
const contract = getContract({
  address: '0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2',
  abi: wagmiAbi,
  client: { wallet: walletClient }
})
const hash = await contract.write.mint([69420], {
  chain: mainnet, // <--- adding this causes typescript to not complain? but doesn't affect the issue.
  gas: 1_234_567n,
})
```
it appears that the `gas` override is being ignored, since the thing is still calling `estimateGas` internally.

how do we override gas limit? thanks in advence.

**Comment by jxom** [comment]
It shouldn’t get overridden. Do you have a minimal repro?

**Reply by firnprotocol** [reply]
you can see the full codebase here https://github.com/firnprotocol/relay but it's not exactly "minimal". can work on one if still necessary. the way that I can "certify" the issue is that I observe two `estimateGas` calls in my Alchemy logs, not one (i.e., per execution of _any one_ among the main if/else branches `req.url === ....` in [index.js](https://github.com/firnprotocol/relay/blob/master/src/index.js)).

**Reply by firnprotocol** [reply]
btw, to clarify, the issue is not anything being "overridden", but rather my _attempt_ to override being ignored.

**Comment by firnprotocol** [comment]
could i please bump this by any chance.... any ideas or thoughts? relevant function call is [here](https://github.com/firnprotocol/relay/blob/9efd4cafb4a18b812bf9904333ecdbc475c486f6/src/index.js#L114-L117).
